technobly/VEXTREME
# Implements Animaction RAM Support

### Problem

Animaction was not working.

### Solution

Enables Animaction support by implementing the write operation back to cartData[] 0x2000 ~ 0x27ff (originally from @rattboi in PR #30 ).  More work had to be done to solve why this wasn't working though; it turned out to be 16-bit writes were not supported in the original code, so operations like STD $ADDR were only writing the first byte.  This is heavily used by Animaction.

### Steps to Test

- Compile for v0.3 (USE_HW=v0.3)
  - Obviously play Animaction!
  - Double check Polar Rescue still works
  - Run the RAM Write test app and verify that all tests PASS: 1:STA INC/ 2:STD INC/ 3:STA $99
- Compile for v0.2 (USE_HW=v0.2)
  - Animaction is still broken, sorry... [you need mods!](https://github.com/technobly/VEXTREME#modifying-v02-to-v03)
  - Double check Polar Rescue still works
  - Run the RAM Write test app and verify that 1:STA INC & 3:STA $99 tests pass, but 2:STD INC test FAILs... sorry... [you need mods!](https://github.com/technobly/VEXTREME#modifying-v02-to-v03)

### References

Closes #30 
Fixes #24
